packages/dashboard/src/components/DashboardFrame/Navigation.tsx

```diff
@@ -12,15 +12,16 @@ import {
 } from '@material-ui/core'
 import { Masks, AccountBalanceWallet, ExpandLess, ExpandMore, Settings } from '@material-ui/icons'
 import { useContext } from 'react'
-import { useRouteMatch } from 'react-router'
+import { useMatch } from 'react-router'
 import { Link, LinkProps } from 'react-router-dom'
-import { Routes } from '../../pages/routes'
+import { RoutePaths } from '../../pages/routes'
 import { DashboardContext } from './context'
 import { MaskNotSquareIcon } from '@dimensiondev/icons'
 import { useDashboardI18N } from '../../locales'
 
 function ListItemLinkUnStyled({ nested, ...props }: LinkProps & ListItemProps & { nested?: boolean; to: string }) {
-    return <MuiListItem button component={Link} selected={!!useRouteMatch(props.to)} {...props} />
+    // @ts-ignore
```

Comment:
When upgrade react-router to v6, `component={Link}` will get a ts error
```
 Property 'component' does not exist on type 'IntrinsicAttributes & { button: true; } & { alignItems?: "flex-start" | "center" | undefined; autoFocus?: boolean | undefined; children?: ReactNode; ... 8 more ...; sx?: SxProps<...> | undefined; } & Omit<...> & CommonProps & Pick<...>'.
```
This is working fine in v5. I'm not sure which community to ask for help here. @Jack-Works

Reply:
Maybe both?

Reply:
And you cannot just ignore the ts error because it is a runtime behavior change.

Reply:
You can write a new component that wraps correctly warp & forwards ref of `component` prop to `as` prop. (https://github.com/ReactTraining/react-router/blob/dev/docs/advanced-guides/migrating-5-to-6.md#rename-link-component-to-link-as-and-navlink-exact-to-navlink-end)

Reply:
I did notice it here, but unfortunately `Link` doesn't have`as` prop

Reply:
```ts
export declare const Link: React.ForwardRefExoticComponent<LinkProps & React.RefAttributes<HTMLAnchorElement>>;
export interface LinkProps extends Omit<React.AnchorHTMLAttributes<HTMLAnchorElement>, 'href'> {
    replace?: boolean;
    state?: State;
    to: To;
}
```